analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.IsNull.cs

```diff
@@ -287,7 +287,9 @@ public void IsNull_TryCast_DownCast()
                 (arg as Exception)?.ToString();
                 Tag("Arg", arg);
                 """, "object arg").Validator;
-        validator.TagValues("Arg").Should().SatisfyRespectively(x => x.Should().BeNull());
+        validator.TagValues("Arg").Should().SatisfyRespectively(
+            x => x.Should().BeNull(),
+            x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
```

Comment:
This is a bit inconsistent now. The conversion isn't learning constraints back to the symbol but isNull does.

Reply:
@pavel-mikula-sonarsource Do you think this is fine?

Reply:
What happens when we know that arg is not null for sure already?
```
arg.ToString(); // Learns NotNull
(arg as Exception)?.ToString();
Tag("Arg", arg);
```

Reply:
Intuitively, the `?` in `(arg as SomeType)?` doesn't declare that users expected `arg` to be null. So it shouldn't learn null. It's not the same as `arg?`

Reply:
Please add UT for it to document it, if not present already.

Reply:
Why does it actually happen for the `arg`? You've already removed the update of the symbol

Reply:
Because `IsNull` calls `TrackedSymbol` on the operand and learns for it. And `TrackedSymbol` unwraps the conversion.